### PR TITLE
F-strings

### PR DESCRIPTION
**Pull request by scopatz**
By request of @asmeurer

### CODE REVIEW

File: tests/test_parser.py [hunk 1]
```diff
@@ -26,6 +26,7 @@ def check_ast(inp, run=True, mode='eval', debug_level=0):
     __tracebackhide__ = True
     # expect a Python AST
     exp = ast.parse(inp, mode=mode)
+    print(pdump(exp))
```
Comment (gforsyth) [inline tests/test_parser.py]: Should this `print` still be here or was this from a debugging run?

**Comment by gforsyth** [comment]
Thanks for putting this in, @scopatz! I tried to think of wacky ways this would break and it seems pretty robust.  Just the one question about the print statement in one of the tests.

Oh, the obfuscated code we'll write...
```console
gil@bad_cat ~/git/xonsh fstr 🐚  a = 'ME'
gil@bad_cat ~/git/xonsh fstr 🐚  ${'HO' + f'{a}'}
'/home/gil'
```

File: tests/test_parser.py (hunk 1 above)
Comment (scopatz) [inline tests/test_parser.py]: Whoops!

**Comment by scopatz** [comment]
Thanks for testing it out @gforsyth. It should be fixed now.

**Comment by astronouth7303** [comment]
Yipeee!!!

Now I get to put in an issue for fg[p]-globs, fr[p]-globs, and fp-strings.

**Comment by gforsyth** [comment]
Thanks, @scopatz !

**Pull request merged**